test/unit/visitor/derivative_original.cpp

```diff
@@ -0,0 +1,54 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include "ast/program.hpp"
+#include "parser/nmodl_driver.hpp"
+#include "test/unit/utils/test_utils.hpp"
+#include "visitors/checkparent_visitor.hpp"
+#include "visitors/derivative_original_visitor.hpp"
+#include "visitors/nmodl_visitor.hpp"
+#include "visitors/symtab_visitor.hpp"
+#include "visitors/visitor_utils.hpp"
+
+using namespace nmodl;
+using namespace visitor;
+using namespace test;
+using namespace test_utils;
+
+using nmodl::parser::NmodlDriver;
+
+
+auto run_cvode_visitor(const std::string& text) {
+    NmodlDriver driver;
+    const auto& ast = driver.parse_string(text);
+    SymtabVisitor().visit_program(*ast);
+    CvodeVisitor().visit_program(*ast);
+
+    return ast;
+}
+
+
+TEST_CASE("Make sure DERIVATIVE block is copied properly", "[visitor][cvode]") {
+    GIVEN("DERIVATIVE block") {
+        std::string nmodl_text = R"(
+            NEURON	{
+                SUFFIX example
+            }
+
+            STATE {x z}
+
+            DERIVATIVE equation {
+                x' = -x + z * z
+                z' = z * x
+            }
+)";
+        auto ast = run_cvode_visitor(nmodl_text);
+        THEN("CVODE block is added") {
+            auto block = collect_nodes(*ast, {ast::AstNodeType::CVODE_BLOCK});
```

Comment:
It's a vector of all `CVODE_BLOCK`s. Hence plural, `blocks` (or `cvode_blocks`); then later you'd be free to use `block = block[0]` (not much use).